lib/matplotlib/axes/_axes.py

```diff
@@ -6581,9 +6579,13 @@ def hist(self, x, bins=None, range=None, density=False, weights=None,
         if len(w) != nx:
             raise ValueError('weights should have the same shape as x')
 
+        input_empty = True
         for xi, wi in zip(x, w):
-            if wi is not None and len(wi) != len(xi):
+            lxi = len(xi)
```

Comment:
Minor nitpick, but for better readability, I would go for
```suggestion
            len_xi = len(xi)
```